source/production/F0.Minesweeper.Logic/Minelayer/FirstEmptyMinelayer.cs

```diff
@@ -2,16 +2,21 @@
 using System.Collections.Generic;
 using System.Linq;
 using F0.Minesweeper.Logic.Abstractions;
+using F0.Minesweeper.Logic.LocationShuffler;
 
 namespace F0.Minesweeper.Logic.Minelayer
 {
 	internal class FirstEmptyMinelayer : IMinelayer
 	{
+		public ILocationShuffler LocationShuffler { get; }
+
+		public FirstEmptyMinelayer(ILocationShuffler locationShuffler)
+			=> LocationShuffler = locationShuffler;
+
 		IReadOnlyCollection<Location> IMinelayer.PlaceMines(IEnumerable<Location> possibleLocations, uint mineCount, Location clickedLocation)
-			=> RemoveClickedLocationArea(possibleLocations.ToList(), clickedLocation)
-				.OrderBy(_ => Guid.NewGuid())
-				.Take((int)mineCount)
-				.ToArray();
+			=> LocationShuffler.ShuffleAndTake(
+				RemoveClickedLocationArea(possibleLocations.ToList(), clickedLocation),
+				(int)mineCount);
```

Comment:
This cast from _uint_ to _int_ is necessary because of the `Take` operation we use inside the shuffler. This could be a - albeit very theoretical - future problem when we introduce custom minefields since one could define a minecount that's greater than an _int_. 
For now maybe we should throw an `ArgumentException` in the constructor of the `Minefield` in case someone tries to generate it with a minecount greater than `int.MaxValue`?

Reply:
Nice one!
Done! But I used an `ArgumentOutOfRangeException`.
See also the [example in the docs](https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception?view=net-5.0).